darc/darc.go

```diff
@@ -863,26 +875,18 @@ func (id Identity) Type() int {
 		return 3
 	case id.EvmContract != nil:
 		return 4
+	case id.DID != nil:
+		return 5
+	case id.TSM != nil:
+		return 6
 	}
 	return -1
 }
 
 // PrimaryIdentity returns true if the identity is a primary identity, which is
 // an identity that is associated with a concrete public/private key.
 func (id Identity) PrimaryIdentity() bool {
-	switch {
-	case id.Darc != nil:
-		return false
-	case id.Ed25519 != nil:
-		return true
-	case id.X509EC != nil:
-		return true
-	case id.Proxy != nil:
-		return true
-	case id.EvmContract != nil:
-		return true
-	}
-	return false
+	return id.Type() != 0
```

Comment:
Does this mean identities other than Darc are always Primary?

Reply:
Yes - this is used to know which identities have counters. It should probably be renamed, but here I just thought it's useless to have this long `switch/case`, when a simple test would do...